charts/spire/charts/spire-server/values.yaml

```diff
@@ -98,9 +98,13 @@ readinessProbe:
   periodSeconds: 5
 
 persistence:
+  # -- What type of volume to use for persistence. Valid options pvc (recommended), hostPath, emptyDir (testing only)
+  type: pvc
```

Comment:
please default to emptyDir.  Not everyone will have a backing store, and one should be able to launch the product to demo it without configuration.

under peristence.type, please structure the per-type values like so

persistence.type.hostpath = "/string/to/path"
persistence.type.pvc.size = whatever
persistence.type.pvc.accessMode = whatever
etc.

If you wish to make persistence.type.pvc into an empty object for maximum user flexibility, I feel that would also be acceptable.

Reply:
For all but the testing case, persistence is needed. By having it default to pvc, it handles the most common case along with making folks think about changing it should they need to. This is the right thing to do IMO.

Why put pvc under type? putting pvc related settings under persistence is the most common api.

Reply:
If we don't put pvc under type, then hostpath will compete with pvc values and one might mix / match where some of the values are ignored and some are not.  By putting it under type, it is clear in the user's values.yaml file if they are mixing and matching, providing visual feedback of the oddity.

Let's say I'm not using Tornjak (sp?) and my keystore, for additional security, is in RAM.  This seems to be a rather "unexceptional" setup.  Then, the only thing I have in the volume is sqlite3, or I'm running with multiple servers, in which case even that's not in the volume.  Which then raises the question, "why are we forcing a PVC on the system, one that will require explicit reconfiguration to avoid?"  Losing a database is nothing, losing a keystore is not the end of the world either, I'm not sure what losing Tornjak entails.

Reply:
The default for upstream spire is sqlite on the equivalent of a hostpath, IE, persistence. One would not ever expect deleting a pod from a node (common on rolling upgrades of Kubernetes) to break your whole spire chain of trust because it has lost its CA cert/key and everything needs rekeying.

Reply:
I'll defer this to the current defaults then.